lib/routes/charge/items/items_list.dart

```diff
@@ -21,8 +22,16 @@ class ItemsList extends StatelessWidget {
         shrinkWrap: true,
         primary: false,
         itemBuilder: (BuildContext context, int index) {
-          return CatalogItem(accountModel, posCatalogBloc, _items[index],
-              _items.length - 1 == index, _addItem);
+          return _filter.isNotEmpty && _containsItem(index) || _filter.isEmpty
+              ? CatalogItem(accountModel, posCatalogBloc, _items[index],
+                  _items.length - 1 == index, _addItem)
+              : Container();
         });
   }
+
+  _containsItem(int index) {
```

Comment:
IMO filtering should be done at a lower level (Repository).
We should add action to SetFilter and the items should be fetched from the repository using this filter.

Reply:
Filtering is being done at repository layer via FilterSales action.
cc13b598dccd453862b5399c9cb6809a4986517c